wire-ios-request-strategy/Tests/Sources/APIs/MessageAPITests.swift

```diff
@@ -0,0 +1,179 @@
+//
+// Wire
+// Copyright (C) 2025 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import XCTest
+@testable import WireRequestStrategy
+
+final class MessageAPITests: XCTestCase {
+
+    var sut: MessageAPI!
+    private var mockHTTPClient: MockHTTPClient!
+    let errorMessage = "test"
+
+    override func setUpWithError() throws {
+        mockHTTPClient = MockHTTPClient()
+        sut = MessageAPIV8(httpClient: mockHTTPClient)
+    }
+
+    override func tearDownWithError() throws {
+        sut = nil
+        mockHTTPClient = nil
+    }
+
+    func test_allMLS_FailureCases() async throws {
```

Comment:
moved the failure tests case here, note we'll have to move this MessageAPI to WireAPI in the future